### ISSUE TITLE
[question] Where to find CachedUniformSampler and CachedCrossBatchSampler implementation

### ISSUE THREAD

**Issue by zexuan-zhou** [NVIDIA-Merlin/models#952]
Hi. Thanks for making the library.

I'm new to the library and I'm looking for the implementation of [CachedUniformSampler](https://github.com/NVIDIA-Merlin/models/blob/60a9ca13ce3975b5a29e6a304d2ead7df193ec6a/tests/common/tf/retrieval/retrieval_utils.py#L84) and [CachedCrossBatchSampler](https://github.com/NVIDIA-Merlin/models/blob/60a9ca13ce3975b5a29e6a304d2ead7df193ec6a/tests/common/tf/retrieval/retrieval_utils.py#L75) but I don't see them in the codebase.

**Comment by rnyak** [comment]
@zexuan-zhou we do currently only support ` mm.InBatchSampler(),`, and do not support  [CachedUniformSampler](https://github.com/NVIDIA-Merlin/models/blob/60a9ca13ce3975b5a29e6a304d2ead7df193ec6a/tests/common/tf/retrieval/retrieval_utils.py#L84) and [CachedCrossBatchSampler](https://github.com/NVIDIA-Merlin/models/blob/60a9ca13ce3975b5a29e6a304d2ead7df193ec6a/tests/common/tf/retrieval/retrieval_utils.py#L75) even though you saw them in the unit tests. We will remove them from the unit test. You can open a feature request for that, or you can do your own custom implementation based on the  old code we had previously: https://github.com/NVIDIA-Merlin/models/blob/v0.5.0/merlin/models/tf/blocks/sampling/cross_batch.py#L26

But please note that this code is not longer supported in our latest releases.

**Comment by rnyak** [comment]
@zexuan-zhou do you have further questions or I can close this ticket? thanks.

**Comment by zexuan-zhou** [comment]
@rnyak Thank you for the update. The implementation of cross_batch looks good. Feel free to close the ticket

A quick question: why do you guys decide on not supporting those two sampling methods anymore?

**Comment by rnyak** [comment]
@zexuan-zhou one reason is that based on our experiments we did not see benefits of these methods. @gabrielspmoreira can provide more info regarding if there was other technical reasons.

**Comment by gabrielspmoreira** [comment]
Hi @zexuan-zhou . In the beginning of last year we did a bunch of experiments with different datasets, comparing in-batch x cross-batch sampling (which are popularity biased) and also uniform sampling for training MF and Two-Tower models. What we found was that both the accuracy and throughput were better with in-batch. 
As we had to do some refactory in the retrieval prediction task API, we ended up keeping only in-batch sampling in the new API as it seemed to be more effective from our experiments. 
But we might revisit this topic again in the future, and check in which conditions cross-batch might be worth
